docs/reference/koin-core/dsl.md

```diff
@@ -110,61 +110,43 @@ Within this option lambda, you can specify the following options:
 
 ### Module Includes (since 3.2)
 
-A new function `includes()` is available in Modules, to let you incldues otehr module for your module and then compose module graph in a better way.
+A new function `includes()` is available in the `Module` class, which lets you compose a module by including other modules in an organized and structured way.
+
+The two prominent use cases of the new feature are:
+- Split large modules into smaller and more focused ones.
+- In modularized projects, it allows you more fine control over a module visibility (see examples below).
 
 How does it work? Let's take some modules, and we include modules in `parentModule`:
 
 ```kotlin
-val childModule1 = module {
-    // other definitions here
-}
-val childModule2 = module {
-    // other definitions here
-}
+// `:feature` module
+internal val childModule1 = module { /* Other definitions here. */ }
+internal val childModule2 = module { /* Other definitions here. */ }
+public val parentModule = module { includes(childModule1,childModule2) }
```

Comment:
Why do you need to add visibility modifier here? (internal, public)

Reply:
That was intentional. It showcases how you can play with visibility modifiers to keep many Koin modules internal while exposing a single "feature module" - that has major implications on how you expose a Gradle module public API.

It is a little hard to explain, as `Module` is overused. Let me give you an example:

<img width="1114" alt="image" src="https://user-images.githubusercontent.com/4348197/157630347-afd2f654-e2b9-4ad8-99b4-43b403002b07.png">

- Green: Gradle modules.
- Blue: Kotlin files.

As you can note above, `:auth` Gradle module contains multiple Koins modules, but only the `AuthModule` is exposed public while all others are now implementation details. That should give fine control over what Koin module is exposed while keeping the not public ones as `included` modules.

What do you think?

Reply:
I see. This is a very good example, but I still wonder if it can go in the "generic" doc 🤔 
It would be a great article for sure.

Reply:
Right, I will remove it for now.

Reply:
👍

Reply:
@arnaudgiuliani changes done.